basyx.submodelservice-core/src/main/java/org/eclipse/digitaltwin/basyx/submodelservice/SubmodelElementIdShortPathParser.java

```diff
@@ -0,0 +1,158 @@
+/*******************************************************************************
+ * Copyright (C) 2023 the Eclipse BaSyx Authors
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ * 
+ * SPDX-License-Identifier: MIT
+ ******************************************************************************/
+package org.eclipse.digitaltwin.basyx.submodelservice;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+import org.eclipse.digitaltwin.basyx.core.exceptions.ElementDoesNotExistException;
+
+/**
+ * Class for parsing an idShortPath
+ * 
+ * @author fried
+ * 
+ */
+public class SubmodelElementIdShortPathParser {
+
+	/**
+	 * Splits an idShortPath
+	 * 
+	 * @param idShortPath
+	 * @return A stack containing all idShortTokens of the idShortPath
+	 * 
+	 */
+	public Stack<String> parsePathTokens(String idShortPath) {
+		try {
+			String splitted[] = splitIdShortPathAtDots(idShortPath);
+			Stack<String> tokenStack = generateTokenStackFromSplittedArray(splitted);
+			return tokenStack;
+		} catch (ElementDoesNotExistException e) {
+			throw new ElementDoesNotExistException(idShortPath);
+		}
+	}
+
+	private static String[] splitIdShortPathAtDots(String idShortPath) {
+		return idShortPath.split("\\.");
+	}
+
+	private static String getIdShortWithoutIndices(String idShort) {
+		return idShort.split("\\[")[0];
+	}
+
+	private static Stack<String> generateTokenStackFromSplittedArray(String[] splitted) {
+		Stack<String> tokenStack = new Stack<>();
+		for (int i = splitted.length - 1; i >= 0; i--) {
+			List<Integer> indices = getAllIndices(splitted[i]);
```

Comment:
Auslagern in eigene Methode z.B. addTokensFromSplittedPart(tokenStack, splitted[i])